util/logtools/encounter_tools.ts

```diff
@@ -0,0 +1,313 @@
+import ContentType from '../../resources/content_type';
+import NetRegexes from '../../resources/netregexes';
+import { UnreachableCode } from '../../resources/not_reached';
+import { default as ZoneInfo } from '../../resources/zone_info';
+import { NetMatches, NetAnyMatches } from '../../types/net_matches';
+import { CactbotBaseRegExp } from '../../types/net_trigger';
+import { LocaleText } from '../../types/trigger';
+import { commonReplacement, syncKeys } from '../../ui/raidboss/common_replacement';
+
+// TODO: add some error checking that a zone has been found before a fight.
+// This can happen on partial logs.
+
+type ZoneEncInfo = {
+  name?: string;
+  zoneId?: number;
+  startLine?: string;
+  endLine?: string;
+  startTime?: Date;
+  endTime?: Date;
+};
+
+type FightEncInfo = {
+  name?: string;
+  zoneId?: number;
+  startLine?: string;
+  endLine?: string;
+  startTime?: Date;
+  endTime?: Date;
+  endType?: string;
+  sealName?: string;
+};
+
+export class EncounterFinder {
+  currentZone: ZoneEncInfo;
+  currentFight: FightEncInfo;
+  currentSeal: string | null;
+  zoneInfo: {
+    readonly exVersion: number;
+    readonly contentType?: number;
+    readonly name: LocaleText;
+    readonly offsetX: number;
+    readonly offsetY: number;
+    readonly sizeFactor: number;
+    readonly weatherRate: number;
+  } | null;
+
+  haveWon: boolean;
+  haveSeenSeals: boolean;
+
+  regex: {
+    changeZone: CactbotBaseRegExp<'ChangeZone'>;
+    cactbotWipe: CactbotBaseRegExp<'GameLog'>;
+    win: CactbotBaseRegExp<'ActorControl'>;
+    wipe: CactbotBaseRegExp<'ActorControl'>;
+    commence: CactbotBaseRegExp<'ActorControl'>;
+    playerAttackingMob: CactbotBaseRegExp<'Ability'>;
+    mobAttackingPlayer: CactbotBaseRegExp<'Ability'>;
+  };
+
+  sealRegexes: Array<CactbotBaseRegExp<'GameLog'>>;
+  unsealRegexes: Array<CactbotBaseRegExp<'GameLog'>>;
+
+  initializeZone(): void {
+    this.currentZone = {};
+  }
+  initializeFight(): void {
+    this.currentFight = {};
+  }
+  constructor() {
+    this.currentZone = {};
+    this.currentFight = {};
+    this.currentSeal = null;
+    // May be null for some zones.
+    this.zoneInfo = null;
+
+    this.haveWon = false;
+    this.haveSeenSeals = false;
+
+    this.regex = {
+      changeZone: NetRegexes.changeZone({ id: '*' }),
+      cactbotWipe: NetRegexes.echo({ line: 'cactbot wipe.*?' }),
+      win: NetRegexes.network6d({ command: '40000003' }),
+      wipe: NetRegexes.network6d({ command: '40000010' }),
+      commence: NetRegexes.network6d({ command: '4000000[16]' }),
+      playerAttackingMob: NetRegexes.ability({ sourceId: '1.{7}', targetId: '4.{7}' }),
+      mobAttackingPlayer: NetRegexes.ability({ sourceId: '4.{7}', targetId: '1.{7}' }),
+    };
+
+    // TODO: consider also doing this with lang? But lang only deals with Regexes???
+    // Sorry, this is quite a bit of a hack.
```

Comment:
Feel free to not make this change here (it's not really a conversion, but an improvement), but I think this whole section could be simplified by using `LocaleNetRegex`.

Reply:
That would definitely be better, but I think you're right that that's best left for another time.